### DISCUSSION TITLE
Zitadel Application slowness in self-hosted setup

### DISCUSSION THREAD

**Discussion by santhosh-stitch** [zitadel/zitadel#8156] [Q&A] [answered]
Zitadel Version : 2.30.0
Database : PostgreSQL-15.4

The API calls made to the Zitadel application are experiencing significant delays, with some requests taking over 25 seconds to process. 
Notably, the endpoint  **/oauth/v2/token** is among those exhibiting prolonged response times. Despite our analysis of the PostgreSQL database, we did not observe any long-running queries.

We observe sequences do not match Warnings in the zitadel application logs. pls find it snippet below.

zitadel[865379]: time="2024-06-18T15:47:38+03:00" level=warning msg="sequences do not match" aggregateType=user caller="/builds/internal/eventstore/handler/crdb/handler_stmt.go:235" currentSeq=191178 prevSeq=191177 projection=projections.login_names2 sequence=191181

**Comment by hifabienne** [comment, marked as answer]
Hei @santhosh-stitch 
We improved a lot in recent versions, as you are still running on 2.30.0 i recommend you to update to a newer version.
My recommendation is at least to the current stable version, you can find it here: https://zitadel.com/docs/support/software-release-cycles-support#stable
